src/codec/RowWriterV2.cpp

```diff
@@ -740,6 +742,27 @@ WriteResult RowWriterV2::write(ssize_t index, const Date& v) noexcept {
     return WriteResult::SUCCEEDED;
 }
 
+WriteResult RowWriterV2::write(ssize_t index, const Time& v) noexcept {
+    auto field = schema_->field(index);
+    auto offset = headerLen_ + numNullBytes_ + field->offset();
+    switch (field->type()) {
+        case meta::cpp2::PropertyType::TIME:
+            buf_[offset] = v.hour;
+            buf_[offset + sizeof(int8_t)] = v.minute;
+            buf_[offset + 2 * sizeof(int8_t)] = v.sec;
+            memcpy(&buf_[offset + 3 * sizeof(int8_t)],
+                   reinterpret_cast<const void*>(&v.microsec),
+                   sizeof(int32_t));
+            break;
+        default:
+            return WriteResult::TYPE_MISMATCH;
+    }
+    if (field->nullable()) {
+        clearNullBit(field->nullFlagPos());
+    }
```

Comment:
Good job,
By the way, what is the range of `time`?

Reply:
00:00:00.0-23:59:59:999999